apps/src/netsim/NetSimVisualization.js

```diff
@@ -432,46 +440,46 @@ NetSimVisualization.prototype.updateVizEntitiesOfType_ = function (
 /**
  * Call kill() on any vizentities that match the given type and don't map to
  * a NetSimEntity in the provided collection.
- * @param {function} vizEntityType
+ * @param {function} vizElementType
  * @param {Array.<NetSimEntity>} entityCollection
  * @private
  */
 NetSimVisualization.prototype.killVizEntitiesOfTypeMissingMatch_ = function (
-    vizEntityType, entityCollection) {
-  this.elements_.forEach(function (vizEntity) {
-    var isCorrectType = (vizEntity instanceof vizEntityType);
+    vizElementType, entityCollection) {
+  this.elements_.forEach(function (vizElement) {
+    var isCorrectType = (vizElement instanceof vizElementType);
     var foundMatch = entityCollection.some(function (entity) {
-      return entity.entityID === vizEntity.id;
+      return vizElement.getCorrespondingEntityID &&
+          entity.entityID === vizElement.getCorrespondingEntityID();
```

Comment:
Yeah, I was hoping to keep the "this maps to a simulation entity" concept out of `VizElement` so it might be possible to map one entity to multiple elements or vice-versa.  Heck, that might have been a better approach to the auto-DNS, letting the simulated router map to both viz nodes.  Things just aren't quite that complex... yet.